### PR TITLE
add collection name as fallback name for nft balances

### PR DESCRIPTION
**Pull request by prettyirrelevant**

### CODE REVIEW

**Comment by kelsos** [comment]
@prettyirrelevant don't forget the documentation update

**Review by yabirgb — COMMENTED** [review]
Is missing documentation and add to the test a check to verify that the information is correctly returned

File: rotkehlchen/tests/api/test_nfts.py [hunk 1]
```diff
@@ -135,3 +135,20 @@ def test_nft_ids_are_unique(rotkehlchen_api_server):
     all_ids = ids_1 + ids_2
     set_of_ids = set(all_ids)
     assert len(all_ids) == len(set_of_ids)
+
+
+@flaky(max_runs=3, min_passes=1)  # all opensea calls have become quite flaky
+@pytest.mark.parametrize('ethereum_accounts', [[TEST_ACC2, TEST_ACC3]])
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_nfts.py:141]: querying just one address is enough

File: rotkehlchen/tests/api/test_nfts.py [hunk 2]
```diff
@@ -135,3 +135,20 @@ def test_nft_ids_are_unique(rotkehlchen_api_server):
...
+@pytest.mark.parametrize('start_with_valid_premium', [True])
+@pytest.mark.parametrize('ethereum_modules', [['nfts']])
+def test_nft_balances_include_collection_name(rotkehlchen_api_server):
+    """Check that nfts balances always include a `collection_name`"""
+    response = requests.get(api_url_for(
+        rotkehlchen_api_server,
+        'nftsbalanceresource',
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_nfts.py:148]: Didn't we have test for this endpoint?

File: rotkehlchen/tests/api/test_nfts.py [hunk 3]
```diff
@@ -135,3 +135,20 @@ def test_nft_ids_are_unique(rotkehlchen_api_server):
...
+    ), json={'async_query': False})
+
+    result = assert_proper_response_with_result(response)
+    for nfts_balances in result.values():
+        for nft_balance in nfts_balances:
+            assert 'collection_name' in nft_balance
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_nfts.py:154]: maybe would be better to query an account like mine `yabir.eth` and check that all the fields are there and are the expected ones (you can search for the ens nft and verify that the output is the expected)

File: rotkehlchen/chain/ethereum/modules/nfts.py [hunk 4]
```diff
@@ -162,6 +163,7 @@ def get_balances(
                     result[address].append({
                         'id': nft.token_identifier,
                         'name': nft.name,
+                        'collection_name': nft.collection.name,
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/nfts.py:166]: Collection is marked as optional in the NFT class. Shouldn't you check if it has collection first?

**Review by yabirgb — COMMENTED** [review]
Left some comments

File: rotkehlchen/tests/api/test_nfts.py (hunk 2 above)
Comment (prettyirrelevant) [reply rotkehlchen/tests/api/test_nfts.py:148]: No test for it

File: rotkehlchen/chain/ethereum/modules/nfts.py [hunk 5]
```diff
@@ -195,7 +199,6 @@ def get_balances(
                         })
                     # Always write detected nfts in the DB to have name and address associated
                     db_data.append((nft.token_identifier, nft.name, None, None, 0, address))  # noqa: E501
-
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/nfts.py:198]: don't remove lines like this. This is a good thing to have to separate code blocks. Please don't do this

**Review by LefterisJP — APPROVED** [review]
LGTM other than the annoying line removal. Someone should fix it in a subsequent PR.

**Pull request merged**